src/erasure-code/ErasureCode.cc

```diff
@@ -195,24 +195,22 @@ int ErasureCode::to_mapping(const map<std::string,std::string> &parameters,
 }
 
 int ErasureCode::to_int(const std::string &name,
-			const map<std::string,std::string> &parameters,
+			ErasureCodeProfile &profile,
 			int *value,
-			int default_value,
+			const std::string &default_value,
 			ostream *ss)
 {
-  if (parameters.find(name) == parameters.end() ||
-      parameters.find(name)->second.size() == 0) {
-    *value = default_value;
-    return 0;
-  }
-  std::string p = parameters.find(name)->second;
+  if (profile.find(name) == profile.end() ||
+      profile.find(name)->second.size() == 0)
+    profile[name] = default_value;
+  std::string p = profile.find(name)->second;
```

Comment:
We could probably avoid  calling find on a profile 3 times with something like: 
ErasureCodeProfile::const_iterator it = profile.find(name);
std::string p;
if(it == profile.end() || it->second.size() == 0) {
  profile[name] = p = default_value;
} else {
  p = it->second;
}